diff --git a/soc/nordic/nrf53/soc.c b/soc/nordic/nrf53/soc.c
index 67eb5dfb67b..34ff4989e16 100644
--- a/soc/nordic/nrf53/soc.c
+++ b/soc/nordic/nrf53/soc.c
@@ -481,13 +481,16 @@ static int nordicsemi_nrf53_init(void)
 		IS_ENABLED(CONFIG_SOC_LFXO_CAP_INT_9PF) ?
 			NRF_OSCILLATORS_LFXO_CAP_9PF :
 			NRF_OSCILLATORS_LFXO_CAP_EXTERNAL);
+
+	nrf_oscillators_lfxo_bypass_set(NRF_OSCILLATORS, true);
 #if !defined(CONFIG_BUILD_WITH_TFM)
 	/* This can only be done from secure code.
 	 * This is handled by the TF-M platform so we skip it when TF-M is
 	 * enabled.
 	 */
 	nrf_gpio_pin_control_select(PIN_XL1, NRF_GPIO_PIN_SEL_PERIPHERAL);
-	nrf_gpio_pin_control_select(PIN_XL2, NRF_GPIO_PIN_SEL_PERIPHERAL);
+	//nrf_gpio_pin_control_select(PIN_XL2, NRF_GPIO_PIN_SEL_PERIPHERAL);
+	nrf_gpio_pin_control_select(PIN_XL2, NRF_GPIO_PIN_SEL_APP);
 #endif /* !defined(CONFIG_BUILD_WITH_TFM) */
 #endif /* defined(CONFIG_SOC_ENABLE_LFXO) */
 #if defined(CONFIG_SOC_HFXO_CAP_INTERNAL)
