diff --git "a/modules/fs/fatfs/include/ffconf.h" "b/modules/fs/fatfs/include/ffconf.h"
index 80c57df..4b923c5 100644
--- "a/modules/fs/fatfs/include/ffconf.h"
+++ "b/modules/fs/fatfs/include/ffconf.h"
@@ -38,7 +38,7 @@
 /* This option switches fast seek function. (0:Disable or 1:Enable) */


-#define FF_USE_EXPAND	0
+#define FF_USE_EXPAND	1
 /* This option switches f_expand function. (0:Disable or 1:Enable) */


@@ -237,7 +237,7 @@
 /  Note that enabling exFAT discards ANSI C (C89) compatibility. */


-#define FF_FS_NORTC		0
+#define FF_FS_NORTC		1
 #define FF_NORTC_MON	1
 #define FF_NORTC_MDAY	1
 #define FF_NORTC_YEAR	2022
diff --git "a/zephyr/subsys/usb/device/class/msc.c" "b/zephyr/subsys/usb/device/class/msc.c"
index 91221a4..b6bd6aa 100644
--- "a/zephyr/subsys/usb/device/class/msc.c"
+++ "b/zephyr/subsys/usb/device/class/msc.c"
@@ -108,7 +108,7 @@ struct CSW {
 /* max USB packet size */
 #define MAX_PACKET	CONFIG_MASS_STORAGE_BULK_EP_MPS

-#define BLOCK_SIZE	512
+#define BLOCK_SIZE	4096
 #define DISK_THREAD_PRIO	-5

 BUILD_ASSERT(MAX_PACKET <= BLOCK_SIZE);
